app/src/org/commcare/activities/InstallArchiveActivity.java

```diff
@@ -205,4 +156,37 @@ public CustomProgressDialog generateProgressDialog(int taskId) {
             return null;
         }
     }
+
+    // Unzip Callbacks
+    @Override
+    public void OnUnzipSuccessful(Integer result) {
+        if (result > 0) {
+            ArchiveFileRoot afr = CommCareApplication.instance().getArchiveFileRoot();
+            String mGUID = afr.addArchiveFile(getTargetFolder());
+
+            String ref = "jr://archive/" + mGUID + "/profile.ccpr";
+
+            Intent i = new Intent(getIntent());
+            i.putExtra(InstallArchiveActivity.ARCHIVE_JR_REFERENCE, ref);
+            setResult(RESULT_OK, i);
+            finish();
+        } else {
+            //assume that we've already set the error message during update, but make it look scary
+            transplantStyle(txtInteractiveMessages, R.layout.template_text_notification_problem);
+        }
+    }
+
+    @Override
+    public void OnUnzipFailure(String cause) {
+        if (!StringUtils.isEmpty(cause)) {

```

Comment:
When we don't have a cause we should still update the mode to error, right?

Reply:
Just noticed below that this was transplanted from the previous behavior, so no need to address if you don't have spare cycles